README
NAME
    Dist::Zilla::PluginBundle::AJGB - Dist::Zilla plugins for AJGB

VERSION
    version 1.103270

SYNOPSIS
        # dist.ini
        [@AJGB]

DESCRIPTION
    This is the plugin bundle for AJGB. It's an equivalent to:

        [@Basic]

        [AutoMetaResources]
        bugtracker.rt = 1
        repository.github = user:ajgb
        homepage = http://search.cpan.org/dist/%{dist}

        [MetaConfig]
        [ModuleBuild]
        [PodCoverageTests]
        [PodSyntaxTests]
        [PkgVersion]
        [AutoVersion]

        [CheckChangesHasContent]
        [CheckExtraTests]
        [CompileTests]
        [EOLTests]
        [NoTabsTests]
        [InstallGuide]
        [KwaliteeTests]
        [PortabilityTests]
        [ReadmeFromPod]

        [Authority]
        authority = cpan:AJGB
        do_metadata = 1

        [PodWeaver]
        config_plugin = @AJGB

AUTHOR
    Alex J. G. Burzyński <ajgb@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Alex J. G. Burzyński
    <ajgb@cpan.org>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Dist-Zilla-PluginBundle-AJGB

2.03   2013-04-07 23:45:50
    NOTICES:
        Just warn about untracked files in Git::Check

2.02   2012-05-09 21:02:37
    Fix tag format
    Commit Changes and README after release

2.01   2012-05-09 20:28:40
    Add missing PkgVersion
    Use plugins for AutoPrereqs

2.00   2012-05-09 20:05:54
    use git plugins
    drop PodWeaver..SeeAlso

1.103270    Tue Nov 23 20:35:01 2010
    Add and use Pod::Weaver::PluginBundle::AJGB
    Use Dist::Zilla::Plugin::Authority 

1.103210    Wed Nov 17 14:52:04 2010
    Config::MVP::Assembler::WithBundles::_add_bundle_contents does not
    support hashrefs, use plain string

0.01        Wed Nov  3 23:51:16 2010
    First version.



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Dist::Zilla plugins for AJGB",
  "dist_author" => [
    "Alex J. G. Burzy\305\204ski <ajgb\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-PluginBundle-AJGB",
  "dist_version" => "2.03",
  "license" => "perl",
  "module_name" => "Dist::Zilla::PluginBundle::AJGB",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Plugin::Authority" => 0,
    "Dist::Zilla::Plugin::AutoMetaResources" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::EOLTests" => 0,
    "Dist::Zilla::Plugin::ExecDir" => 0,
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "Dist::Zilla::Plugin::Git::Check" => 0,
    "Dist::Zilla::Plugin::Git::Commit" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::Git::Push" => 0,
    "Dist::Zilla::Plugin::Git::Tag" => 0,
    "Dist::Zilla::Plugin::InstallGuide" => 0,
    "Dist::Zilla::Plugin::KwaliteeTests" => 0,
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => 0,
    "Dist::Zilla::Plugin::Manifest" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaConfig" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::ModuleBuild" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::NoTabsTests" => 0,
    "Dist::Zilla::Plugin::PkgVersion" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::ReadmeFromPod" => 0,
    "Dist::Zilla::Plugin::ShareDir" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::Synopsis" => 0,
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Moose" => 0,
    "Pod::Weaver::Config::Assembler" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Test::More" => 0
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Pod/Weaver/PluginBundle/AJGB.pm
use strict;
use warnings;
package Pod::Weaver::PluginBundle::AJGB;
BEGIN {
  $Pod::Weaver::PluginBundle::AJGB::AUTHORITY = 'cpan:AJGB';
}
{
  $Pod::Weaver::PluginBundle::AJGB::VERSION = '2.03';
}
#ABSTRACT: Pod::Weaver configuration for AJGB

use Pod::Weaver::Config::Assembler;


sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub mvp_bundle_config {
    return (
        [ '@AJGB/CorePrep',     _exp('@CorePrep'),  {} ],

        [ '@AJGB/Encoding',     _exp('-Encoding'),  {} ],

        [ '@AJGB/Name',         _exp('Name'),       {} ],
        [ '@AJGB/Version',      _exp('Version'),    {} ],


        [ '@AJGB/Prelude',      _exp('Region'),     { region_name   => 'prelude'        } ],

        [ '@AJGB/Synopsis',     _exp('Generic'),    { header        => 'SYNOPSIS'       } ],
        [ '@AJGB/Description',  _exp('Generic'),    { header        => 'DESCRIPTION'    } ],
        [ '@AJGB/Overview',     _exp('Generic'),    { header        => 'OVERVIEW'       } ],


        [ 'ATTRIBUTES',         _exp('Collect'),    { command       => 'attr'           } ],
        [ 'METHODS',            _exp('Collect'),    { command       => 'method'         } ],
        [ 'FUNCTIONS',          _exp('Collect'),    { command       => 'func'           } ],
        [ 'TYPES',              _exp('Collect'),    { command       => 'type'           } ],

        [ '@AJGB/Leftovers',    _exp('Leftovers'),  {} ],

        [ '@AJGB/Postlude',     _exp('Region'),     { region_name   => 'postlude'       } ],

        [ '@AJGB/Authors',      _exp('Authors'),    {} ],
        [ '@AJGB/Legal',        _exp('Legal'),      {} ],
    );
}

1;


__END__
=pod

=encoding utf-8

=head1 NAME

Pod::Weaver::PluginBundle::AJGB - Pod::Weaver configuration for AJGB

=head1 VERSION

version 2.03

=head1 DESCRIPTION

Pod::Weaver configuration used by L<Dist::Zilla::PluginBundle::AJGB>.

=for Pod::Coverage mvp_bundle_config

=head1 AUTHOR

Alex J. G. Burzyński <ajgb@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2012 by Alex J. G. Burzyński <ajgb@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist::Zilla plugins for AJGB",
  "AUTHOR" => "Alex J. G. Burzy\305\204ski <ajgb\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-AJGB",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::PluginBundle::AJGB",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::Authority" => 0,
    "Dist::Zilla::Plugin::AutoMetaResources" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::EOLTests" => 0,
    "Dist::Zilla::Plugin::ExecDir" => 0,
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "Dist::Zilla::Plugin::Git::Check" => 0,
    "Dist::Zilla::Plugin::Git::Commit" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::Git::Push" => 0,
    "Dist::Zilla::Plugin::Git::Tag" => 0,
    "Dist::Zilla::Plugin::InstallGuide" => 0,
    "Dist::Zilla::Plugin::KwaliteeTests" => 0,
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => 0,
    "Dist::Zilla::Plugin::Manifest" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaConfig" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::ModuleBuild" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::NoTabsTests" => 0,
    "Dist::Zilla::Plugin::PkgVersion" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::ReadmeFromPod" => 0,
    "Dist::Zilla::Plugin::ShareDir" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::Synopsis" => 0,
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Moose" => 0,
    "Pod::Weaver::Config::Assembler" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Test::More" => 0
  },
  "VERSION" => "2.03",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



